File: jcarrollonlinev4.backend/Models/ForumThreadEntries/ThreadEntryDetailItemsModel.cs
using jcarrollonlinev4.backend.Models.HierarchyNode;

namespace jcarrollonlinev4.backend.Models.ForumThreadEntries
{
    public class ThreadEntryDetailItemsModel : ThreadEntriesModelBase
    {
        public int NumberOfReplies { get; set; }

        public IEnumerable<HierarchyNodesModel<ThreadEntryDetailsItemModel>>? ForumThreadEntries { get; set; }
    }
}

File: jcarrollonlinev4.frontend/src/models/ForumThreadEntries/ThreadEntryDetailItemsModel.ts
import { HierarchyNodesModel } from "../HierarchyNode/HierarchyNodeModel";
import { ThreadEntryModel } from "./ThreadEntryModel";

export interface ThreadEntryDetailItemsModel {
  NumberOfReplies: number;
  ForumThreadEntries: HierarchyNodesModel<ThreadEntryModel>;
}
